airbytehq/airbyte
# Python Ruff support (config only)

This can safely merge ahead of the large auto-fix PR:

- https://github.com/airbytehq/airbyte/pull/29853

This PR adds configuration for Ruff, allowing developers to applying their own auto-fixes and to see what changes or issues Ruff would find, without making it a breaking change and without impacting any existing workstreams.

## To test in VS Code

Install the VS Code extension for Ruff: https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff


## To test with the command line

Install Ruff with pip or pipx:

```console
pipx install ruff
```

Run Ruff on the entire monorepo:

```console
ruff check --fix .
```

OR run Ruff on any subproject's directory:

```console
ruff check --fix airbyte-ci/connectors/pipelines
ruff check --fix airbyte-cdk/python
```
